klippy/extras/display/menu.py

```diff
@@ -989,6 +1009,17 @@ def key_event(self, key, eventtime):
         self.display.request_redraw()
 
     # Collection of manager class helper methods
+    # Possible boolean values in the configuration.
+    BOOLEAN_STATES = {
+        '1': True,
+        'yes': True,
+        'true': True,
+        'on': True,
+        '0': False,
+        'no': False,
+        'false': False,
+        'off': False,
+    }
```

Comment:
Please don't add this.  Booleans should be read from the config with `config.getboolean()`.  Adding code to support multiple ways of doing the same thing does not significantly reduce user costs, but it does increase the overhead for maintaining code, maintaining documentation, and diagnosing error reports.

Reply:
I cannot use the `config.getboolean()` method because menu items can also be created manually.
Then the config variable is not config file anymore but is dict class / `MenuConfig` class 

Like this 
```py
item = self.manager.menuitem_from({
	'type': 'command',
	'name': self.manager.asliteral(fname),
	'gcode': "\n".join(gcode)
})

self._itemBack = self.manager.menuitem_from({
	'type': 'command',
	'name': '..',
	'gcode': '{menu.back()}'
})
```
This was the reason why I needed duplicating some `config file` methods for the menu system.
The `config.getboolean()` only allows to read from config sections but I need to
evaluate values in code too.
```py
def eval_enable(self):
	context = super(MenuInput, self).get_context()
	return self.manager.asbool(self._enable_tpl.render(context)).
```
the `config.getboolean()` was an excellent example how to interpret boolean values.

Reply:
I'm not sure I understand.  However, for run-time menu evaluation of "enabled", one could use `operator.truth(ast.literal_eval(some_text))`.

-Kevin

Reply:
Thanks. I'll check it.